extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -9,69 +9,195 @@
     private static final String VAULT_SECRET_DATA_PATH = "data";
     private static final String VAULT_SECRET_METADATA_PATH = "metadata";
-    private static final String CALL_UNSUCCESSFUL_ERROR_TEMPLATE = "[Hashicorp Vault] Call unsuccessful: %s";
+    private static final String CALL_UNSUCCESSFUL_ERROR_TEMPLATE = "Call unsuccessful: %s";
+    private static final String TOKEN_LOOK_UP_SELF_PATH = "/v1/auth/token/lookup-self";
+    private static final String TOKEN_RENEW_SELF_PATH = "/v1/auth/token/renew-self";
     private static final int HTTP_CODE_404 = 404;
+    private static final String TIME_FORMAT = "%02dh:%02dm:%02ds";
+
     @NotNull
-    private final HashicorpVaultClientConfig hashicorpVaultConfig;
+    private final HashicorpVaultConfig hashicorpVaultConfig;
     @NotNull
     private final EdcHttpClient httpClient;
     @NotNull
+    private final ScheduledExecutorService scheduledExecutorService;
+    @NotNull
     private final ObjectMapper objectMapper;
+    @NotNull
+    private final Monitor monitor;
 
-    HashicorpVaultClient(@NotNull HashicorpVaultClientConfig hashicorpVaultConfig, @NotNull EdcHttpClient httpClient,
-                         @NotNull ObjectMapper objectMapper) {
+    HashicorpVaultClient(@NotNull HashicorpVaultConfig hashicorpVaultConfig,
+                         @NotNull EdcHttpClient httpClient,
+                         @NotNull ObjectMapper objectMapper,
+                         @NotNull ScheduledExecutorService scheduledExecutorService,
+                         @NotNull Monitor monitor) {
         this.hashicorpVaultConfig = hashicorpVaultConfig;
         this.httpClient = httpClient;
         this.objectMapper = objectMapper;
+        this.scheduledExecutorService = scheduledExecutorService;
+        this.monitor = monitor;
+    }
+
+    Result<HealthCheckResponse> doHealthCheck() {
+        var requestUri = getHealthCheckUrl();
+        var request = httpGet(requestUri);
+
+        try (var response = httpClient.execute(request)) {
+            var code = response.code();
+            var responseBody = Objects.requireNonNull(response.body()).string();
+            var responsePayload = objectMapper.readValue(responseBody, HealthCheckResponsePayload.class);
+            var healthCheckResponse = HealthCheckResponse.Builder
+                    .newInstance()
+                    .code(code)
+                    .payload(responsePayload)
+                    .build();
+            return Result.success(healthCheckResponse);
+        } catch (IOException e) {
+            return Result.failure(e.getMessage());
+        }
+    }
+
+    void inspectToken(TokenLookUpResponsePayloadToken token) {
+        if (token.isRootToken()) {
+            monitor.warning("Root token detected. Don't use root tokens in production environment!");
+        } else {
+            if (token.isRenewable()) {
+                monitor.info("Token is renewable");
```

Comment:
What purpose does logging this at INFO level serve? INFO level is for important information and I don't think this should be logged at all (not even at DEBUG).